assets/app/data/services/upgrade/upgrade-admin.factory.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+
+    angular
+        .module('crowbarData.upgrade')
+        .factory('upgradeUpgradeAdminFactory', upgradeUpgradeAdminFactory);
+
+    upgradeUpgradeAdminFactory.$inject = ['$q', '$http'];
+    /* @ngInject */
+    function upgradeUpgradeAdminFactory($q, $http) {
+        return {
+            getAdminUpgrade: getAdminUpgrade,
+            getAdminUpgradeStatus: getAdminUpgradeStatus
+        };
+
+        function getAdminUpgrade() {
+            var requestOptions = {
+                method: 'POST',
+                url: '/api/upgrade7/admin-upgrade'
+            };
+
+            return $http(requestOptions);
+        }
+
+        function getAdminUpgradeStatus() {
```

Comment:
Yes, I think a new provider (`crowbarFactory`?) can be created to encapsulate this.